pkg/loki/format_query_handler.go

```diff
@@ -22,6 +23,12 @@ func formatQueryHandler() http.HandlerFunc {
 			statusCode = http.StatusBadRequest
 			status = "invalid-query"
 			errStr = err.Error()
+			if strings.Contains(errStr, "parse error at line 1") && strings.Contains(errStr, "unexpected IDENTIFIER") {
```

Comment:
I appreciate what you're trying to do here, but I think this is too generic; an unexpected identifier could still be encountered in a LogQL query.

I do think, though, that it would be very nice to return more context to users when errors are encountered. I would be in favour of adding an additional field to the response JSON where we could insert links and/or help text under certain conditions.

Reply:
OK - do you think we could expose that field in the LogCLI output as well? 

I'm actually a bit confused about this because what I see when I get the error back from the server is `text/plain` output but I can't figure out where that is being generated - I looked at all instances in the source tree where we send back text/plain responses and they don't seem to make sense. 

```
$ curl -v -u "$USER:$(cat disk-tokens/grafana-cloudaccesspolicy-token)" https://logs-prod-006.grafana.net/loki/api/v1/query_range\?direction\=BACKWARD\&end\=1693289918339003000\&limit\=30\&query\=segment\&start\=1693286318339003000
> GET /loki/api/v1/query_range?direction=BACKWARD&end=1693289918339003000&limit=30&query=segment&start=1693286318339003000 HTTP/1.1
> Host: logs-prod-006.grafana.net
> Authorization: Basic <redacted>
> User-Agent: curl/8.2.1
> Accept: */*
>
< HTTP/1.1 400 Bad Request
< Content-Length: 65
< Content-Type: text/plain; charset=utf-8
< Date: Thu, 31 Aug 2023 19:59:09 GMT
< Vary: Accept-Encoding
<
parse error at line 1, col 1: syntax error: unexpected IDENTIFIER
```

Reply:
I opened #10414 to discuss